File: articles/security-failures-as-algorithmic-failures/java/SecurityFailures.java
public class SecurityFailures {
    public static void main(String[] args) {
        double[] weights = {0.08,0.10,0.10,0.10,0.08,0.08,0.08,0.08,0.08,0.08,0.06,0.05,0.03};
        double score = 0.0;
        for (double w : weights) {
            score += w * 0.65;
        }
        System.out.printf("%.3f%n", score * 100.0);
    }
}

File: articles/security-failures-as-algorithmic-failures/fortran/security_failures.f90
program security_failures
  implicit none
  real :: weights(13) = (/0.08,0.10,0.10,0.10,0.08,0.08,0.08,0.08,0.08,0.08,0.06,0.05,0.03/)
  real :: score
  score = 100.0 * sum(weights * 0.65)
  print '(F8.3)', score
end program security_failures
